crates/subspace-node/src/bin/subspace-node.rs

```diff
@@ -501,8 +501,19 @@ fn main() -> Result<(), Error> {
                         ))
                     })?;
 
+                    // if is dev, use the known key ring to start relayer
```

Comment:
Hm... just like we're not assuming executors, I think we shouldn't assume relayers either, just document what to specify in documentation. We have https://github.com/subspace/subspace/blob/main/docs/development.md but it doesn't mention anything about execution layer yet.

Reply:
This wont be run if there is no secondary chain or core domain is running. If they do and this is dev, then we run relayers for each domain since `dev` chainspec have them included.

Reply:
What if you want to run two nodes locally that each have their own relayer? This makes it impossible to do so unfortunately.

Reply:
No that should not be a problem unless both the nodes are started with same known keyring.  The one problem I see is if we exhaust all the known keyrings or want override the relayer_id with some other address, then that override is ignored. I have thought about it and felt that we dont want to run that many nodes locally in dev mode anyway and also avoid unnecessary check in the code.

Reply:
I mean this feels like a workaround that makes us unable to test certain things for the benefit of skipping a single CLI argument.

There is already `--alice` and others in Substrate, can we maybe use them instead? I'm not sure if relayer has to be an executor on the secondary chain, but if it is, then that would make sense to me.

Reply:
>  I mean this feels like a workaround that makes us unable to test certain things for the benefit of skipping a single CLI argument.

Not sure I understand why you see this change as workaround. We ideally need to pass `relayer_id` for every domain we initialise. If there is no relayer_id, then relayer wont run for that domain. This PR wont affect that behavior.

As for the dev env, we are injecting a known keyring as relayer address if alice, bob, ... are passed with the args just like how we are authoring blocks in primary or execution layer. Except the change here is visible unlike Executor since Executors check with SyncKeystore for availble keys and continue author as known keyrings are already injected.

Reply:
We're not injecting any specific identity into primary chain to produce blocks just because `--dev` is specified. To me this change is **invisible** to the user and other cases is visible because it requires an explicit argument unless I'm missing something.

Reply:
Not in the primary chain sure. But on the executor we cross check Known Keys from Keystore with executors registered on the runtime. 
So from here - https://github.com/paritytech/substrate/blob/master/client/cli/src/params/shared_params.rs#L36-L37
When we run `--dev` alone, the `--alice` is implicit and is available to the keystore. So Executors in dev chain dont need to excliptly inject keys to the keystore to sign bundles with.

So for the relayers, we dont need a keystore since we dont sign the messages relayers are submitting since the reward model is different. So When we want to run the relayer on dev chain, we are implictly just like executors for alice and other known keys take the known keyring public address and start a relayer.

Reply:
I see, thanks for explaining that, now it makes much more sense now!

So the way I see it would ideally work then is the following:
* `SubstrateCli::create_configuration` for secondary chain should just be absorbed by `Configuration::new` (basically let it take `secondary_chain_cli` and `tokio_handler` as arguments, we don't need three arguments here)
* `Configuration::new` will first create `service_config` with `SubstrateCli::create_configuration`
* Then `secondary_chain_cli.run.relayer_id` will be used for executor identity and if not provided will check `.is_dev()` and fall back to `get_keyring` to obtain the behavior you want here
* Rename `Configuration` to something less generic because it is a bit misleading to see it in primary chain's node like that (at least as an import alias)